src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -479,7 +480,7 @@ void QgsRelationReferenceWidget::init()
         QgsFeatureIterator fit = mReferencedLayer->getFeatures();
         while ( fit.nextFeature( ft ) )
         {
-          for ( int i = 0; i < mFilterComboBoxes.count() - 1; ++i )
+          for ( int i = 0; i < mFilterFields.count() - 1; ++i )
```

Comment:
I already hit this kind of problem in this widget (`mFilterComboBoxes` and `mFilterFields` not in sync). To be sure that it won't happen again, I used the min index instead:

``` c++
const int count = std::min( mFilterComboBoxes.size(), mFilterFields.size() );
for ( int i = 0; i < count; i++ )
{
  ...
}
```

See https://github.com/qgis/QGIS/blob/master/src/gui/editorwidgets/qgsrelationreferencewidget.cpp#L270-L276